### ISSUE TITLE
Integration tests on loss functions

### ISSUE THREAD

**Issue by mathpluscode** [DeepRegNet/DeepReg#622]
## Subject of the code / test

Learnt from MONAI that we can write some tests to train very small stuff on each loss, to make sure it's differentiable and the loss range is reasonable. As some loss should be between -1 and 0 and we do not expect any values outside.

This requirement has helped us to find a bug today and therefore it might be good for us?

**Comment by NMontanaBrown** [comment]
And maybe regression tests?
